packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -75,10 +83,17 @@ function preload(
       seen[dep] = true
       const isCss = dep.endsWith('.css')
       const cssSelector = isCss ? '[rel="stylesheet"]' : ''
-      // @ts-ignore check if the file is already preloaded by SSR markup
-      if (document.querySelector(`link[href="${dep}"]${cssSelector}`)) {
+      const isBaseRelative = !!importerUrl
+
+      // check if the file is already preloaded by SSR markup
+      if (isBaseRelative) {
+        // When isBaseRelative is true then we have `importerUrl` and `dep` is
+        // already converted to an absolute URL by the `assetsURL` function
+        if (absoluteUrls.has(dep)) return
```

Comment:
I think this does not work when `<link rel="preload" as="style" href="foo.css">` exists. (`<link rel="stylesheet" href="foo.css">` should be injected but will be skipped.)